darshan-util/pydarshan/darshan/experimental/plots/plot_opcounts.py

```diff
@@ -0,0 +1,157 @@
+    stdio_vals = [0, 0, 0, 0, 0, 0, 0]
+
+
+    # TODO: change to report.summary
+    if 'agg_ioops' in dir(report):
+        print("Summarizing... agg_ioops")
+        report.agg_ioops()
+    else:
+        print("Can not create summary, agg_ioops aggregator is not registered with the report clase.")
+
+
+    mods = report.summary['agg_ioops']
+
+    # Gather POSIX
+    if 'POSIX' in mods:
+        #posix_record = backend.log_get_posix_record(log)
+        #posix = dict(zip(backend.counter_names("POSIX"), posix_record['counters']))
+        posix = mods['POSIX']
+
+        posix_vals = [
+            posix['POSIX_READS'],
+            posix['POSIX_WRITES'],
+            posix['POSIX_OPENS'],
+            posix['POSIX_STATS'],
+            posix['POSIX_SEEKS'],
+            0, # faulty? posix['POSIX_MMAPS'],
+            posix['POSIX_FSYNCS'] + posix['POSIX_FDSYNCS']
+        ]
+
+    # Gather MPIIO
+    if 'MPI-IO' in mods:
+        #mpiio_record = backend.log_get_mpiio_record(log)
+        #mpiio = dict(zip(backend.counter_names("mpiio"), mpiio_record['counters']))
+
+        mpiio = mods['MPI-IO']
+
+        mpiind_vals = [
+            mpiio['MPIIO_INDEP_READS'],
+            mpiio['MPIIO_INDEP_WRITES'],
+            mpiio['MPIIO_INDEP_OPENS'],
+            0, # stat
+            0, # seek
+            0, # mmap
+            0, # sync
+        ]
+
+        mpicol_vals = [
+            mpiio['MPIIO_COLL_READS'],
+            mpiio['MPIIO_COLL_WRITES'],
+            mpiio['MPIIO_COLL_OPENS'],
+            0, # stat
+            0, # seek
+            0, # mmap
+            mpiio['MPIIO_SYNCS']
+        ]
+
+    # Gather Stdio
+    if 'STDIO' in mods:
+        #stdio_record = backend.log_get_stdio_record(log)
+        #stdio = dict(zip(backend.counter_names("STDIO"), stdio_record['counters']))
+
+        stdio = mods['STDIO']
+
+        stdio_vals = [
+            stdio['STDIO_READS'],
+            stdio['STDIO_WRITES'],
+            stdio['STDIO_OPENS'],
+            0, # stat
+            stdio['STDIO_SEEKS'],
+            0, # mmap
+            stdio['STDIO_FLUSHES']
+        ]
+
+
+
+    def as_csv():
+        text = ""
+        text += ','.join(labels) +  ',Layer' + "\n"
+        text += ','.join(str(x) for x in posix_vals) + ',POSIX' + "\n"
+        text += ','.join(str(x) for x in mpiind_vals) + ',MPIIND' + "\n"
+        text += ','.join(str(x) for x in mpicol_vals) + ',MPICOL' + "\n"
+        text += ','.join(str(x) for x in stdio_vals) + ',STDIO' + "\n"
+
+        return text
+
+
+
+    print(as_csv())
+
+
+    x = np.arange(len(labels))  # the label locations
+    width = 0.15  # the width of the bars
+
+    fig, ax = plt.subplots()
+    rects1 = ax.bar(x - width/2 - width, posix_vals, width, label='POSIX')
+    rects2 = ax.bar(x - width/2, mpiind_vals, width, label='MPI-IO Indep.')
+    rects3 = ax.bar(x + width/2, mpicol_vals, width, label='MPI-IO Coll.')
+    rects4 = ax.bar(x + width/2 + width, stdio_vals, width, label='STDIO')
```

Comment:
My impression from early discussions with Shane is that we may want to focus on a single module to start with for the report--perhaps `DXT_POSIX`. Anyway, establishing which module we want to focus on first probably makes sense.

I haven't checked if your aggregator code has handling to prevent combining duplicates between i.e., `POSIX` and `DXT_POSIX`; that kind of stuff was starting to get tricky to think about.

Reply:
We should probably discuss this in more detail to make sure we're all on the same page. I think we want to use DXT data for the heatmap plots in particular, since it provides finer-grained details that make heatmaps both more informative and easier to develop. For the other plots that we typically provide on darshan-job-summary, DXT is probably not what we want to use as a starting point...plots like the operation count plots, average I/O time per-process, etc. probably should be prototyped using our traditional instrumentation modules that provide summaries (POSIX makes most sense to start with since almost all logs have some POSIX data), rather than using fine-grained DXT traces.

Reply:
I've been using POSIX yeah. Direct access to pandas dataframe data is a bit more natural there if you don't need DXT anyway.